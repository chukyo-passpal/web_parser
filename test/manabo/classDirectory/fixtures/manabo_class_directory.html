<script>
    $(function () {
        // event setup
        $('.li-directories').find('.glyphicon-folder-toggle').on('click', function (e) {
            e.preventDefault();
            $(this).closest('li').find('ul').slideToggle('fast');
            if ($(this).hasClass('glyphicon-folder-open')) {
                $(this).removeClass('glyphicon-folder-open').addClass('glyphicon-folder-close');
            } else {
                $(this).removeClass('glyphicon-folder-close').addClass('glyphicon-folder-open');
            }
            setTimeout(function () { saveDirectoryOpenState(); }, 300);
        }).css('cursor', 'pointer');

        $('.a-open-directory-form').on('click', function (e) {
            e.preventDefault();
            openDirectoryModal($(this));
        });

        $('.a-remove-directory').on('click', function (e) {
            e.preventDefault();
            removeDirectory();
        });

        $('.a-copy-directory').on('click', function (e) {
            e.preventDefault();
            copyDirectory();
        });

        $('.a-open-contents').on('click', function (e) {
            e.preventDefault();
            glexa.sessionStorage.setItem('class_id', 85131);
            glexa.sessionStorage.setItem('directory_id', $(this).attr('directory_id') ? $(this).attr('directory_id') : 0);
            saveDirectoryOpenState($(this).attr('directory_id'));
            glexa.loadClassDirectories(85131, $(this).attr('directory_id'));
            glexa.loadClassNews(85131, $(this).attr('directory_id'));
            glexa.loadClassContents(85131, $(this).attr('directory_id'));
        });

        $('.checkbox-directories').on('change', function (e) {
            setDirectoryEditMode($(this).val());
        });

        // first load
        setDirectoryEditMode();
        initDirectoryOpenState();
    });

    function initDirectoryOpenState() {
        openState = eval('[' + glexa.localStorage.getItem('directory_open_state_85131') + ']');
        if (!openState) {
            openState = [];
        }


        $('.li-directories').each(function () {
            if (openState.indexOf(parseInt($(this).attr('directory_id'))) < 0) {
                $(this).find('.glyphicon-folder-toggle').removeClass('glyphicon-folder-open').addClass('glyphicon-folder-close');
                $(this).find('ul').hide();
            }
        });
    }

    function saveDirectoryOpenState(directoryId) {
        var openState = [];
        if (directoryId) {
            openState.push(directoryId);
        }
        $('.li-directories').each(function () {
            if ($(this).find('ul').is(':visible')) {
                openState.push($(this).attr('directory_id'));
            }
        });
        glexa.localStorage.setItem('directory_open_state_85131', openState);
    }

    function setDirectoryEditMode(mode) {
        resetDirectoryUI();
        if (!mode) {
            mode = glexa.sessionStorage.getItem('directory_edit_mode');
            $('input[name=directory_edit_' + mode + ']').prop('checked', true);
        }

        if ($('input[name=directory_edit_' + mode + ']').prop('checked')) {
            var mode = $('.checkbox-directories:checked').val();
            $('.checkbox-directories').prop('checked', false);
            $('input[name=directory_edit_' + mode + ']').prop('checked', true);
            $('header,.btn,.panel').children().css({ opacity: 0.5 });
            $('.div-panel-directory').children().css({ opacity: 1 });

            if (mode == 'sort') {
                glexa.setSort({
                    element: '.x-directory-sort',
                    axis: 'y'
                });
            }
            if (mode == 'move') {
                glexa.setDrag({
                    dragElement: '.x-directory-drag',
                    dropElement: '.x-directory-drop',
                    helper: helperDirectory,
                    onDropped: function (e, ui) {
                        params = {
                            directory_id: ui.draggable.attr('directory_id'),
                            target_directory_id: $(e.target).attr('directory_id')
                        }
                        glexa.ajax({
                            action: 'teacher_ajax_class_directory_list_move_accept',
                            params: params,
                            onSuccess: function () {
                                glexa.loadClassDirectories(85131, ui.draggable.attr('directory_id'));
                            }
                        });
                    }
                });
            }
            glexa.sessionStorage.setItem('directory_edit_mode', mode);
            $('input[name=directory_edit_' + mode + ']').parent('label').addClass("active");
        } else {
            glexa.sessionStorage.removeItem('directory_edit_mode');
        }
    }

    function resetDirectoryUI() {
        $('header,.btn,.panel').children().css({ opacity: 1 });
        glexa.destroySort({ element: '.x-directory-sort' });
        glexa.destroyDrag({ dragElement: '.x-directory-drag', dropElement: '.x-directory-drop' });
    }

    function helperDirectory() {
        return '<fieldset style="min-width:300px;margin-top:15px;">' + $(this).find('.x-directory-name').html() + '</fieldset>';
    }

    function openDirectoryModal(object) {
        params = {
            class_id: 85131,
        }
        if (object.attr('directory_id')) {
            params.directory_id = object.attr('directory_id');
        }
        if (object.attr('parent_id')) {
            params.parent_id = object.attr('parent_id');
        }
        glexa.openRemoteModal({
            action: 'teacher_modal_class_directory_form',
            params: params
        });
    }

    function setDirectorySorable(directoryIds) {
        glexa.ajax({
            action: 'teacher_ajax_class_directory_list_sort_accept',
            params: {
                class_id: 85131,
                directory_ids: directoryIds
            },
            onSuccess: function (res) {
            }
        });
    }

    function removeDirectory() {
    }

    function copyDirectory() {
        callback = function (e) {
            var is_limited = $(e.target).hasClass('btn-common-confirm-modal-yes') ? 1 : 0
            glexa.confirm('ディレクトリのコピーを開始してよろしいですか？<br />時間がかかる場合がありますのでそのままお待ちください', function (e) {
                glexa.ajax({
                    action: 'teacher_ajax_class_directory_list_copy_accept',
                    params: {
                        class_id: 85131,
                        directory_id: 0,
                        is_limited: is_limited
                    },
                    onSuccess: function () {
                        glexa.loadClassDirectories(85131, 0);
                    }
                });
            })
            return false;
        }

        glexa.confirm('公開設定の期間もコピーしますか？', callback, '', callback);
    }
</script>

<style>
    .class-directory .a-class-name,
    .class-directory span.span-class-name {
        padding-left: 0;
    }
</style>
<div class="panel panel-default sp-margin-bottom-none div-panel-directory">
    <div class="panel-heading text-center">ディレクトリ</div>
    <div class="panel-body">
        <div class="class-directory">
            <ul>
                <li>
                    <span class="title_normal">
                        <div class="class-top-directory">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131" directory_id="0">
                                <span class="span-class-name"><b>アルゴリズムとデータ構造２</b></span>
                            </div>
                        </div>
                    </span>

                    <ul>
                        <li sortable_ids="1214705" class="x-directory-drag" directory_id="1214705">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214705">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214705">
                                        <span class="x-directory-name">第1回 マージと分割統治法</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214706" class="x-directory-drag" directory_id="1214706">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214706">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214706">
                                        <span class="x-directory-name">第2回 クイックソート</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214707" class="x-directory-drag" directory_id="1214707">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214707">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214707">
                                        <span class="x-directory-name">第3回 構造体、ポインターによるデータ構造の定義</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214708" class="x-directory-drag" directory_id="1214708">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214708">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214708">
                                        <span class="x-directory-name">第4回 リスト構造</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214709" class="x-directory-drag" directory_id="1214709">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214709">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214709">
                                        <span class="x-directory-name">第5回 リスト構造の実現</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214710" class="x-directory-drag" directory_id="1214710">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214710">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214710">
                                        <span class="x-directory-name">第6回 スタックとキュー</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214711" class="x-directory-drag" directory_id="1214711">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214711">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214711">
                                        <span class="x-directory-name">第7回 スタックとキューの実現</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214712" class="x-directory-drag" directory_id="1214712">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214712">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214712">
                                        <span class="x-directory-name">第8回 グラフ</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214713" class="x-directory-drag" directory_id="1214713">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214713">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214713">
                                        <span class="x-directory-name">第9回 グラフの基本探索アルゴリズム</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214714" class="x-directory-drag" directory_id="1214714">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214714">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214714">
                                        <span class="x-directory-name">第10回 グラフの基本探索アルゴリズムの実現</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214715" class="x-directory-drag" directory_id="1214715">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214715">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214715">
                                        <span class="x-directory-name">第11回 洗練された探索アルゴリズム</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214716" class="x-directory-drag" directory_id="1214716">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214716">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214716">
                                        <span class="x-directory-name">第12回 木構造と二分探索木</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214717" class="x-directory-drag" directory_id="1214717">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214717">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214717">
                                        <span class="x-directory-name">第13回 二分探索木の実現と基本操作</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214718" class="x-directory-drag" directory_id="1214718">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214718">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214718">
                                        <span class="x-directory-name">第14回 ヒープ</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                        <li sortable_ids="1214719" class="x-directory-drag" directory_id="1214719">
                            <div class="x-content-drop x-directory-drop x-news-drop" class_id="85131"
                                directory_id="1214719">
                                <span class="glyphicon glyphicon-folder-toggle glyphicon-folder-open"
                                    aria-hidden="true"></span>
                                <span class="x-drag">
                                    <a href="#" class="a-open-contents" directory_id="1214719">
                                        <span class="x-directory-name">第15回 ヒープの実現と基本操作</span>

                                    </a>
                                </span>
                            </div>
                            <ul>
                            </ul>
                        </li>
                    </ul>
                </li>
            </ul>
        </div><!--class-directory-->
    </div><!--panel-body-->


    <ul class="margin-bottom-none">
        <div id="div-directory_form"></div>
    </ul>
</div>
