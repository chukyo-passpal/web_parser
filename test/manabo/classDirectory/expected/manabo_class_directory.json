{
  "classId": "85131",
  "className": "アルゴリズムとデータ構造２",
  "directories": [
    {
      "directoryId": "1214705",
      "title": "第1回 マージと分割統治法"
    },
    {
      "directoryId": "1214706",
      "title": "第2回 クイックソート"
    },
    {
      "directoryId": "1214707",
      "title": "第3回 構造体、ポインターによるデータ構造の定義"
    },
    {
      "directoryId": "1214708",
      "title": "第4回 リスト構造"
    },
    {
      "directoryId": "1214709",
      "title": "第5回 リスト構造の実現"
    },
    {
      "directoryId": "1214710",
      "title": "第6回 スタックとキュー"
    },
    {
      "directoryId": "1214711",
      "title": "第7回 スタックとキューの実現"
    },
    {
      "directoryId": "1214712",
      "title": "第8回 グラフ"
    },
    {
      "directoryId": "1214713",
      "title": "第9回 グラフの基本探索アルゴリズム"
    },
    {
      "directoryId": "1214714",
      "title": "第10回 グラフの基本探索アルゴリズムの実現"
    },
    {
      "directoryId": "1214715",
      "title": "第11回 洗練された探索アルゴリズム"
    },
    {
      "directoryId": "1214716",
      "title": "第12回 木構造と二分探索木"
    },
    {
      "directoryId": "1214717",
      "title": "第13回 二分探索木の実現と基本操作"
    },
    {
      "directoryId": "1214718",
      "title": "第14回 ヒープ"
    },
    {
      "directoryId": "1214719",
      "title": "第15回 ヒープの実現と基本操作"
    }
  ]
}